### DISCUSSION TITLE
Viomi V8 (Xiaomi Mi Robot Vacuum Mop Pro)

### DISCUSSION THREAD

**Discussion by jsantosdias** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#383] [Q&A] [answered]
Hello!

I'm trying to control Xiaomi Mi Robot Vacuum Mop Pro with HASS and I reached 
lovelace-xiaomi-vacuum-map-card 

It seems to work, but I can't use rooms. Is it possible to someone help me on what should I do to make it work?

Thank you!
JD

**Comment by PiotrMachowski** [comment]
What integration do you use to control your vacuum?

**Comment by jsantosdias** [comment]
I have Xiaomi  #MIoT.

**Comment by PiotrMachowski** [comment, marked as answer]
https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/issues/251

**Comment by jsantosdias** [comment]
Ok, thank you very much for the reply!

So what I understood is that lovelace xiaomi vacuum doesn’t work with zones with Viomi V8. Is this correct?

If I want to use the code, what file should I write in?

Thank you!

João Dias

**Reply by PiotrMachowski** [reply]
> So what I understood is that lovelace xiaomi vacuum doesn’t work with zones with Viomi V8. Is this correct?

It should work, but you would need to figure out a correct config by yourself.

> If I want to use the code, what file should I write in?

You have to use it in card's configuration (in yaml editor when editing the card).

--- 

Generally speaking Xiaomi MIoT is pretty hard to work with, I suggest migrating to [this integration](https://github.com/tykarol/home-assistant-viomi-vacuum-v8), it should be working out of the box
